### PR TITLE
Run3-sim118A Try to replace getByLabel with token in SimCalorimetry/EcalElectronicsEmulation

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to replace getByLabel with token in SimCalorimetry/EcalElectronicsEmulation

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc [hunk 1]
```diff
@@ -166,7 +166,9 @@ string EcalSimRawData::getExt() const {
   }
 }
 
-void EcalSimRawData::genFeData(string basename, int iEvent, const vector<uint16_t> adcCount[nEbEta][nEbPhi]) const {
+void EcalSimRawData::genFeData(std::string basename,
```
Comment (civanch) [inline SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc:169]: @bsunanda , why not std::string&  ?

File: SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc [hunk 2]
```diff
@@ -63,44 +57,50 @@ EcalSimRawData::EcalSimRawData(const edm::ParameterSet &params) {
...
   // Event counter:
   ++iEvent;
 
   if (xtalVerbose_ | tpVerbose_) {
-    cout << "=================================================================="
-            "====\n"
-         << " Event " << iEvent << "\n"
-         << "------------------------------------------------------------------"
-            "----\n";
+    std::cout << "=================================================================="
```
Comment (civanch) [inline SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc:82]: @bsunanda , why std::cout and no end of line? I mean "/n" as a last symbol not always complete the line.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc [hunk 3]
```diff
@@ -63,44 +57,50 @@ EcalSimRawData::EcalSimRawData(const edm::ParameterSet &params) {
...
     writeMode_ = bigEndian;
   } else {
     writeMode_ = ascii;
   }
+
+  eeSrFlagToken_ = consumes<EESrFlagCollection>(edm::InputTag(srDigiProducer_, eeSrFlagCollection_));
+  ebSrFlagToken_ = consumes<EBSrFlagCollection>(edm::InputTag(srDigiProducer_, ebSrFlagCollection_));
+  ebDigisToken_ = consumes<EBDigiCollection>(edm::InputTag(digiProducer_, ebDigiCollection_));
+  trigPrimDigisToken_[0] = consumes<EcalTrigPrimDigiCollection>(edm::InputTag(tpProducer_, tcpDigiCollection_));
+  trigPrimDigisToken_[1] = consumes<EcalTrigPrimDigiCollection>(edm::InputTag(tpProducer_, tpDigiCollection_));
```
Comment (thomreis) [inline SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc:74]: Better to use enums for the array indices.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc [hunk 4]
```diff
@@ -555,31 +555,34 @@ void EcalSimRawData::getTp(const edm::Event &event, const std::string &collName,
...
 
       tcp[iTtEta0][iTtPhi0] = tp[tp.sampleOfInterest()].raw();
 
       if (tpVerbose_) {
-        if (tcp[iTtEta0][iTtPhi0] != 0)  // print non-zero values only
-          cout << collName << (collName.empty() ? "" : " ") << "TP(" << setw(2) << iTtEta0 << "," << iTtPhi0 << ") = "
-               << "0x" << setw(4) << tcp[iTtEta0][iTtPhi0] << "\tcmssw indices: " << tp.id().ieta() << " "
-               << tp.id().iphi() << "\n";
+        if (tcp[iTtEta0][iTtPhi0] != 0) {  // print non-zero values only
+          std::string collName = (type == 0) ? tcpDigiCollection_ : tpDigiCollection_;
```
Comment (perrotta) [inline SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc:577]:
```suggestion
          std::string collName = (type == EcalSimRawData::tcp) ? tcpDigiCollection_ : tpDigiCollection_;
```
(Very minor point, not worth preventing this to get merged, but perhaps you'll like to take it into account in further cleanups)

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**